### PR TITLE
Use stack in ci

### PR DESCRIPTION
**Pull request by abailly**
Simplifies `.travis.yml`  with the use of stack and caching of dependencies.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,62 +1,67 @@
...
+  - sudo apt-get install -qq emacs24 emacs24-el
+
+  ## Emacs 24.4
+  - wget http://ftpmirror.gnu.org/emacs/emacs-24.4.tar.xz
+  - tar -xf emacs-24.4.tar.xz
+  - cd emacs-24.4
+  - ./configure --prefix=/usr/local/emacs-24.4 --with-xpm=no --with-gif=no
+  - make
+  - sudo make install
+  - cd ..
```
Comment (david-christiansen) [inline .travis.yml]: Shouldn't we now test on Emacs 25.3 and 24.5, and cache the resulting directories?

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
 
-BATCHEMACS=$(EMACS) --batch --no-site-file -q -eval '(add-to-list (quote load-path) "${PWD}/")' -eval '(require (quote package))' -eval '(add-to-list (quote package-archives) (quote ("melpa" . "http://melpa.org/packages/")) t)' -eval '(package-initialize)'
+BATCHEMACS=$(EMACS) --batch --no-site-file -q \
+	-eval '(add-to-list (quote load-path) "${PWD}/")' \
+	-eval '(require (quote package))' \
+	-eval '(add-to-list (quote package-archives) (quote ("melpa" . "http://melpa.org/packages/")) t)' \
+	-eval '(package-initialize)'
 
-BYTECOMP = $(BATCHEMACS) -eval '(progn (require (quote bytecomp)) (setq byte-compile-warnings t) (setq byte-compile-error-on-warn t))' -f batch-byte-compile
+BYTECOMP = $(BATCHEMACS) -eval '(progn (require (quote bytecomp)) (setq byte-compile-warnings t) (setq byte-compile-error-on-warn nil))' -f batch-byte-compile
```
Comment (david-christiansen) [inline Makefile]: Rather than not erroring on byte-compiler warnings, shouldn't we fix them?

File: Makefile [hunk 3]
```diff
@@ -32,16 +36,16 @@ OBJS =	idris-commands.elc		\
...
+test: getdeps
 	$(BATCHEMACS) -L . -l ert -l idris-tests.el -f ert-run-tests-batch-and-exit
 
 clean:
 	-rm -f $(OBJS)
 	-rm -f test-data/*ibc
 
 getdeps:
-	$(BATCHEMACS) -eval '(progn (package-refresh-contents) (package-install (quote prop-menu)))'
+	$(BATCHEMACS) -eval '(progn (package-refresh-contents) (unless (package-installed-p (quote prop-menu)) (package-install (quote prop-menu)) ))'
```
Comment (david-christiansen) [inline Makefile]: The space at the end of the line between closing parens is not good Lisp style. Ending parens should be in one gigantic pile :-)))))))

**Comment by david-christiansen** [comment]
Seeing this green check mark makes me feel good :-) Thanks!

File: .travis.yml (hunk 1 above)
Comment (abailly) [reply .travis.yml]: Yes, but I was planning to do that in a later step. Actually I would like to have a clear separation of builds with a "matrix" of emacs versions versus idris versions...

File: Makefile (hunk 2 above)
Comment (abailly) [reply Makefile]: Indeed! Same answer than before: baby steps

**Comment by abailly** [comment]
@david-christiansen If you are ok with my comments, I will let you merge and will work towards addressing those issues.

**Comment by abailly** [comment]
Fix #468

**Comment by david-christiansen** [comment]
I'm busy moving this week and have basically no bandwidth of the brain. Please use your best judgement and merge if you feel it's time.

Thanks!

**Pull request merged**